File: models/Show.js
class Show {

    /**
     * @param {Object} obj 
     */
    constructor(obj) {
        this.id = obj.id;
        this.title = obj.title;
        this.poster = obj.poster;
        this.kinds = obj.kinds.split(";");
        this.favorite = obj.favorite;
        this.missing = obj.missing;
        this.duration = obj.duration;
        this.addedAt = obj.added_at;
        this.country = obj.country;
        this.watch = obj.continue;
        this.seasons = obj.seasons;
    }
}

module.exports = Show;

File: models/show.go
package models

import (
	"anothapp-v3/entities"
	"time"
)

type PostShowDto struct {
	Id int `binding:"required" json:"id"`
}

type ShowInfo struct {
	ID       uint      `json:"id"`
	Title    string    `json:"title"`
	Poster   string    `json:"poster"`
	Duration uint      `json:"duration"`
	Seasons  uint      `json:"seasons"`
	Country  string    `json:"country"`
	Favorite bool      `json:"favorite"`
	Continue bool      `json:"continue"`
	AddedAt  time.Time `json:"addedAt"`
}

func NewShowInfo(userShow entities.UserShow) *ShowInfo {
	return &ShowInfo{
		ID:       userShow.ShowID,
		Title:    userShow.Show.Title,
		Poster:   userShow.Show.Poster,
		Duration: userShow.Show.Duration,
		Seasons:  userShow.Show.SeasonsNumber,
		Country:  userShow.Show.Country,
		Favorite: userShow.Favorite,
		Continue: userShow.Continue,
		AddedAt:  userShow.AddedAt,
	}
}
